src/main/java/hudson/remoting/UserRequest.java

```diff
@@ -90,6 +90,17 @@ public UserRequest(Channel local, Callable<?,EXC> c) throws IOException {
         this.classLoaderProxy = RemoteClassLoader.export(cl, local);
     }
 
+    @Override
+    public void checkIfCanBeExecutedOnChannel(Channel channel) throws IOException {
+        // Default check for all requests
+        super.checkIfCanBeExecutedOnChannel(channel);
+        
+        // We also do not want to run UserRequests when the channel is being closed
+        if (channel.isClosingOrClosed()) {
+            throw new IOException("The request cannot be executed on channel " + channel + ". The channel is closing down or has closed down");
```

Comment:
Unfortunately this low-level method is being called not only within `Channel#call`, but also from `Request#call()` directly (also a public API). Hence I had to add a check there.

I could just remove checks in `Channel#call()`, but in such case some logic in requests will be invoked (e.g. UserRequest constructor, export object recorder, etc.). I would prefer to fail fast there